magento/inventory
# Products created with WEBAPI without quantity are automatically assigned to default source.

### Description (*)

- Source Items and Source Item Configurations are not removed when product is deleted, therefore a new product with same SKU will inherit previously deleted product source items data (Fixed in  #2795 )
- Products created with webapi without stock data are assigned to default source.

### Manual testing scenarios (*)
- Install Magento with MSI
- Create new Source Stores->Inventory->Sources (Name: Second, code: second), specify the address, Save.
- Create a new product through the backend. Name: Simple Product1, sku: test111, price: 100, assign sources: second, qty: 100. Save
- Go back to the product grid. Note the quantity per source: "Second: 100".
- Mark the product in the grid and delete it.
- Create a new integration: System -> Extensions -> Integrations. Enter the name, your password, and all privileges. Save. Activate. Copy the Access token.
- Go to the Swagger: http://site.local/swagger/ . Specify the token, apply it.
Select catalogProductRepositoryV1 -> POST - > /V1/products, Try it out.
Specify the request body: 
```json
{
 "product": {
    "sku": "test111",
    "name": "Simple Product2",
    "attribute_set_id": 4,
    "price": 100,
    "type_id": "simple"
 }
```

- Note that the sku is the same as in #3. Click Execute. The product must be successfully created (code 200)
- Go back to the backend in the product grid.

EXPECTED RESULTS

The new product is not assigned to sources and has no quantity.

ACTUAL RESULTS

The new product is assigned with two sources - the default with qty 0 and the second with qty 100 from the deleted product.

### Questions or comments

This is issue is not reproducible if the product is created on backend. However the source items data are not removed until a new product is created that match the  previously deleted product's SKU, then old source items data are removed on fly.

### Contribution checklist (*)
